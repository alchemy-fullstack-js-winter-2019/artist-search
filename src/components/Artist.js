import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { getArtist } from '../services/musicBrainzApi';

export default class Artist extends PureComponent {
  constructor(props) {
    super(props);
  }

  state = {
    artist: '',
    works: []
  }

  fetchWorks = () => {
    getArtist(this.props.match.params.id)
      .then(res => {
        console.log('response', res);
        this.setState({
          artist: res.artist,
          works: res.works
        });
      });
  }

  componentDidMount() {
    console.log('mounted');
    this.fetchWorks();
  }

  render() {
    const worksList = this.state.works.map(work => {
      return (
        <Work work={work} key={work.id}/>
      );
    });
    return (
      <>
      <h1>{this.state.artist}</h1>
      <ul>
        {worksList}
      </ul>
      </>
    );
  }
}

Artist.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  })
};


