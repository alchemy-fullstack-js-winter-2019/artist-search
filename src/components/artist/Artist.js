import React, { PureComponent } from 'react';


export default class Artist extends PureComponent {

    render() {
        return (
            <h1> Artist component</h1>
        );
    }
}
