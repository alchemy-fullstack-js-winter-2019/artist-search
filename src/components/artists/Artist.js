import React from 'react';
import PropTypes from 'prop-types';

function Artist({ artist }) {
  const { name } = artist;
  return (
    <>
      <li>{name}</li>
    </>
  );
}

Artist.propTypes = {
  artist: PropTypes.object
};

export default Artist;
