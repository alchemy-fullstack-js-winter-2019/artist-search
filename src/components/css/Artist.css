@import url('https://fonts.googleapis.com/css?family=Staatliches');

.artist {
  h1 {
    margin: 0;
    font-family: Staatliches;
    padding: 10px;
    text-align: center;
  }
}
