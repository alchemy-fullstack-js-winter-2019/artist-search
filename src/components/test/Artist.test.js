import React from 'react';
import Artist from '../Artist';
import renderer from 'react-test-renderer';

jest.mock('../../services/musicBrainzApi.js');

describe('Artist', () => {
  it('matches a snapshot', () => {
    const match = { params:{ id: '57f73e51-5b95-485b-9102-bc7bd23584f7' } };
    console.log('WOW****', match);
    const tree = renderer.create(
      <Artist match={match}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
