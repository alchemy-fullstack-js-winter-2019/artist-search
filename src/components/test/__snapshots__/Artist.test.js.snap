// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist matches a snapshot 1`] = `
<div
  className="artist"
>
  <h1>
    
  </h1>
  <ul />
</div>
`;
